realm/src/main/java/io/realm/Realm.java

```diff
@@ -413,6 +413,7 @@ public static Realm getInstance(File writeableFolder, String fileName, byte[] ke
     }
 
     private static Realm create(File writableFolder, String filename, byte[] key) {
+        assertValidRealmPath(writableFolder, filename);
```

Comment:
I would avoid the word assert here. Maybe "check" would be better. Just a detail of course